provision-phase-1/Jenkinsfile

```diff
@@ -0,0 +1,34 @@
+pipeline {
+    agent {
+        dockerfile {
+            filename 'Dockerfile.build'
+
+            // Install awscli for uploading the built tarballs to S3.
+            additionalBuildArgs "--build-arg=packages=awscli"
```

Comment:
Not sure it really matters, but the jenkins job has a `BRANCH` parameter for the OS and the `Dockerfile` supports a `branch` build arg, but it's not passed through here.

Reply:
In theory we should be building in an OS branch that matches the code branch. In practice I don't think it really matters, since all that is being called in the build is `tar` and `sha256sum`, and PAYG repos have dev and stable branches almost always flush with one another.

If we change it here we should also change it for the other jobs in `payg.yaml`

Reply:
I suggest `additionalBuildArgs "--build-arg=packages=awscli --build-arg=branch=${params.BRANCH}"`. Then the OS branch will match the git branch and the build deps are more likely to match.

Reply:
I think it's not running `tar` that's an issue, but rather running `meson` since it will try find all the dependencies for the whole project.

Reply:
Yep, I know. That was why I also mentioned that the branches are almost always flush, so I don't think we'll hit problems with dependencies.

In any case, I agree that passing the branch is the right thing to do, sorry for going on a unnecessary tangent. I'll update this PR.